src/main/java/seedu/address/logic/commands/AddLogCommand.java

```diff
@@ -0,0 +1,194 @@
+
+        // add to address book
+        model.setPerson(personToEdit, addedLogPerson);
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        return new CommandResult(MESSAGE_ADD_LOG_SUCCESS);
+    }
+
+    /**
+     * Creates a {@code Person} with the details of {@code personToEdit}, with logs modified by
+     * {@code editPersonLogsDescriptor}.
+     *
+     * @throws CommandException if {@code addLogDescriptor} results in an invalid {@code Log}
+     *                          being created.
+     */
+    private static Person createAddedLogPerson(Person personToEdit, AddLogDescriptor addLogDescriptor)
+            throws CommandException {
+        requireAllNonNull(personToEdit, addLogDescriptor);
+        Name name = personToEdit.getName();
+        Phone phone = personToEdit.getPhone();
+        Email email = personToEdit.getEmail();
+        Address address = personToEdit.getAddress();
+        Set<Tag> tags = personToEdit.getTags();
+        List<Log> updatedLogs = addLogDescriptor.getLogs(personToEdit); // main logic encompassed here
+        return new Person(name, phone, email, address, tags, updatedLogs);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof AddLogCommand)) {
+            return false;
+        }
+
+        // cast
+        AddLogCommand a = (AddLogCommand) other;
+        return this.index.equals(a.index)
+                && this.addLogDescriptor.equals(a.addLogDescriptor);
+    }
+
+    @Override
+    public String toString() {
+        return "Index: " + this.index.toString() + "\nContent:\n" + this.addLogDescriptor.toString();
+    }
+
+    /**
+     * Stores the details of the edited log to edit a person's logs with, as well as the person's
+     * original details.
+     */
+    public static class AddLogDescriptor {
+
+        private String newTitle;
+        private String newDescription;
+
+        /**
+         * Constructs a new {@code AddLogDescriptor} object.
+         */
+        public AddLogDescriptor() {
+            this.newTitle = null;
+            this.newDescription = null;
+        }
+
+        public void setNewTitle(String newTitle) {
+            this.newTitle = newTitle;
+        }
+
+        public void setNewDescription(String newDescription) {
+            this.newDescription = newDescription;
+        }
+
+        /**
+         * Returns true if title has been edited.
+         */
+        public boolean isTitleEdited() {
+            return this.newTitle != null;
+        }
+
+        /**
+         * Returns a list of {@code Log} objects that include the {@code Person}'s original logs
+         * as well as the new logs.
+         */
+        public List<Log> getLogs(Person personToEdit) throws CommandException {
+            try {
```

Comment:
Maybe change the name of the function to suit its functionality? Since this function adds a new log to the old log list.